src/consensus/pbft/libbyz/client_proxy.h

```diff
@@ -211,7 +211,7 @@ bool ClientProxy<T, C>::send_request(
   if (current_outstanding.fetch_add(1) >= Max_outstanding)
   {
     current_outstanding.fetch_sub(1);
-    LOG_FAIL << "Too many outstanding requests, rejecting!" << std::endl;
+    LOG_FAIL_FMT("Too many outstanding requests, rejecting!");
```

Comment:
Better to avoid `!`?

Reply:
For what it's worth, these are short and reasonable: https://docs.microsoft.com/en-us/windows/win32/debug/error-message-guidelines?redirectedfrom=MSDN#style-considerations

https://docs.microsoft.com/en-us/windows/win32/uxguide/mess-error?redirectedfrom=MSDN is also insightful, even if a lot of it is only really relevant to pop up boxes.

In particular:

```
In contrast to the previous bad examples, good error messages have:

A problem. States that a problem occurred.
A cause. Explains why the problem occurred.
A solution. Provides a solution so that users can fix the problem.
Additionally, good error messages are presented in a way that is:

Relevant. The message presents a problem that users care about.
Actionable. Users should either perform an action or change their behavior as the result of the message.
User-centered. The message describes the problem in terms of target user actions or goals, not in terms of what the code is unhappy with.
Brief. The message is as short as possible, but no shorter.
Clear. The message uses plain language so that the target users can easily understand problem and solution.
Specific. The message describes the problem using specific language, giving specific names, locations, and values of the objects involved.
Courteous. Users shouldn't be blamed or made to feel stupid.
Rare. Displayed infrequently. Frequently displayed error messages are a sign of bad design.
```